daemon/daemon.go

```diff
@@ -623,6 +623,10 @@ func NewDaemon(config *config.Config, registryService registry.Service, containe
 		os.Setenv("TMPDIR", realTmp)
 	}
 
+	if config.ContainerdNamespace == "" {
+		config.ContainerdNamespace = ContainersNamespace
```

Comment:
Looks like we need some validation on this;

```bash
dockerd --containerd-namespace="hello world"
```

```
DEBU[2018-05-19T19:06:41.987274386Z] processing event stream                       module=libcontainerd namespace="hello world"
ERRO[2018-05-19T19:06:41.987788322Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.987921624Z] processing event stream                       module=libcontainerd namespace="plugins.hello world"
ERRO[2018-05-19T19:06:41.988186851Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==plugins.hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="plugins.hello world"
DEBU[2018-05-19T19:06:41.988142306Z] processing event stream                       module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.988608030Z] processing event stream                       module=libcontainerd namespace="plugins.hello world"
ERRO[2018-05-19T19:06:41.988938267Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.989253238Z] processing event stream                       module=libcontainerd namespace="hello world"
ERRO[2018-05-19T19:06:41.989891739Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.990024270Z] processing event stream                       module=libcontainerd namespace="hello world"
ERRO[2018-05-19T19:06:41.989891331Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==plugins.hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="plugins.hello world"
DEBU[2018-05-19T19:06:41.990252468Z] processing event stream                       module=libcontainerd namespace="plugins.hello world"
ERRO[2018-05-19T19:06:41.990646964Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==plugins.hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="plugins.hello world"
DEBU[2018-05-19T19:06:41.990748019Z] processing event stream                       module=libcontainerd namespace="plugins.hello world"
ERRO[2018-05-19T19:06:41.991305511Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.991406990Z] processing event stream                       module=libcontainerd namespace="hello world"
ERRO[2018-05-19T19:06:41.991860574Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.992128312Z] processing event stream                       module=libcontainerd namespace="hello world"
ERRO[2018-05-19T19:06:41.991378514Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==plugins.hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="plugins.hello world"
ERRO[2018-05-19T19:06:41.992490201Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.992811583Z] processing event stream                       module=libcontainerd namespace="hello world"
^CERRO[2018-05-19T19:06:41.993449302Z] failed to get event                           error="rpc error: code = Unknown desc = subscription error: failed parsing subscription filters: filters: parse error: [namespace==hello >| |< world,topic~=|^/tasks/|]: unexpected input:  : invalid argument" module=libcontainerd namespace="hello world"
DEBU[2018-05-19T19:06:41.993673609Z] processing event stream                       module=libcontainerd namespace="hello world"
INFO[2018-05-19T19:06:41.993796788Z] Processing signal 'interrupt'                
DEBU[2018-05-19T19:06:41.994040940Z] daemon configured with a 15 seconds minimum shutdown timeout 
DEBU[2018-05-19T19:06:41.994092214Z] start clean shutdown of all containers with a 15 seconds timeout... 
```

However; this starts:

```bash
dockerd --containerd-namespace="test,namespace==foo"
```

But then running a container succeeds, but produces an error;;

```
docker: Error response from daemon: namespace validation: namespace "test,namespace==foo" must match ^[A-Za-z][A-Za-z0-9]+(:?[-]+[A-Za-z][A-Za-z0-9]+)*(?:[.](?:[A-Za-z][A-Za-z0-9]+(:?[-]+[A-Za-z][A-Za-z0-9]+)*))*$: invalid argument.
```

Reply:
Validation happens here : https://github.com/moby/moby/pull/36914/files#diff-4eeac9ba6fec6b0677037c6f8735d16bR555